File: openapi-test-java-server-spring/testapi/model/BaseEntity.java
package testapi.model;

import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.validation.constraints.NotNull;
import java.time.OffsetDateTime;
import java.util.Optional;

public record BaseEntity(
  @JsonProperty("createdAt") @NotNull OffsetDateTime createdAt,
  @JsonProperty("id") @NotNull String id,
  @JsonProperty("updatedAt") Optional<OffsetDateTime> updatedAt
) {
  public BaseEntity withCreatedAt(OffsetDateTime createdAt) {
    return new BaseEntity(createdAt, id, updatedAt);
  };

  public BaseEntity withId(String id) {
    return new BaseEntity(createdAt, id, updatedAt);
  };

  public BaseEntity withUpdatedAt(Optional<OffsetDateTime> updatedAt) {
    return new BaseEntity(createdAt, id, updatedAt);
  };
}

File: openapi-test-kotlin-server-spring-client-mp/testapi/model/BaseEntity.kt
package testapi.model

import com.fasterxml.jackson.annotation.JsonProperty
import jakarta.validation.constraints.NotNull
import java.time.OffsetDateTime
import java.util.Optional

data class BaseEntity(
  @field:JsonProperty("createdAt") @NotNull val createdAt: OffsetDateTime,
  @field:JsonProperty("id") @NotNull val id: String,
  @field:JsonProperty("updatedAt") val updatedAt: Optional<OffsetDateTime>
)